### PR TITLE
Fix for testUserDefinedEndpoint

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFSalesforceAction.m [hunk 1]
```diff
@@ -99,24 +99,26 @@ - (id)contentFromData:(NSData*)data fromResponse:(NSHTTPURLResponse*)response er
     NSError *responseError = nil;
     id content = [super contentFromData:data fromResponse:response error:&responseError];
 
-    if (content && !responseError) {
+    if (!responseError) {
```
Comment (wmathurin) [inline libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFSalesforceAction.m]: In the case of a 404 (which is what the test does), we have a nil content and no responseError. If we don't enter the if, the code will get back a nil object and no error.

**Comment by bhariharan** [comment]
:+1: 

That big merge is going to get a bit more messy now ;-)

File: libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFSalesforceAction.m (hunk 1 above)
Comment (wmathurin) [inline libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFSalesforceAction.m]: I wonder some of the code in this method in CSFSalesforceAction belongs in the super class?

**Comment by kchitalia** [comment]
:+1:

File: libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFSalesforceAction.m (hunk 1 above)
Comment (bhariharan) [inline libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFSalesforceAction.m]: Some of it, like the ```errorMsg``` logic is specific to ```Salesforce API```, that's why I left it here.

**Pull request merged**